File: 김준서/Week_12/P155651_호텔대실.py
def solution(book_time):
    answer = 0
    con = [0 for _ in range(24*60+11)]
    for start, end in book_time:
        start = list(map(int, start.split(":")))
        start = start[0]*60+start[1]
        end = list(map(int, end.split(":")))
        end = end[0]*60+end[1]
        con[start]+=1
        con[end+10]-=1
    for i in range(24*60+10):
        con[i+1]+=con[i]
        answer = max(answer, con[i+1])
    return answer

File: 윤상우/Week_12/P155651_호텔대실.java
package temp;
import java.util.*;

public class P155651_호텔대실 {
    public int solution(String[][] book_time) {
        int answer = 1;
        int len = book_time.length;
        int [][] book_int = new int [len][2];

        for(int i=0; i<len; i++){
            int start = Integer.parseInt(book_time[i][0].substring(0,2)) * 60 + Integer.parseInt(book_time[i][0].substring(3));
            int end = Integer.parseInt(book_time[i][1].substring(0,2)) * 60 + Integer.parseInt(book_time[i][1].substring(3));

            book_int[i] = new int []{start,end};
        }

        Arrays.sort(book_int, (o1,o2) -> (o1[0] - o2[0]));

        for(int i=0; i<len; i++){
            System.out.println(book_int[i][0] + ", " + book_int[i][1]);
        }

        PriorityQueue<Integer> que = new PriorityQueue<>();
        que.add(book_int[0][1]+10);

        for(int i=1; i<len; i++){
            int in = book_int[i][0];
            int out = que.poll();

            if(out > in){
                answer++;
                que.add(out);
                que.add(book_int[i][1] + 10);
            }else{
                que.add(book_int[i][1] + 10);
            }
        }

        return answer;
    }
}
